prow/cmd/sinker/main.go

```diff
@@ -69,42 +66,27 @@ func main() {
 		logrus.WithError(err).Fatal("Error starting config agent.")
 	}
 
-	clusterConfig, err := rest.InClusterConfig()
-	if err != nil {
-		logrus.WithError(err).Fatal("Error loading cluster config.")
-	}
+	clusterConfigs, defaultContext, err := kube.LoadClusterConfigs(o.buildClusterKubeconfig, o.buildCluster)
+	defaultConfig := clusterConfigs[defaultContext]
 
-	pjclient, err := pjclientset.NewForConfig(clusterConfig)
+	pjclient, err := pjclientset.NewForConfig(&defaultConfig)
```

Comment:
Based on this area of code, this will mean that longer term we'll provide all prow components with a single kubeconfig file that defines multiple contexts, e.g.:

- default (i.e. Prow cluster)
- build-cluster
- build-cluster-trusted

It'd be good to keep the door open for potentially supporting a cluster-per-file model or similar too maybe?
This would make it easier to dynamically register new clusters (but we could punt this into there's actually a user requirement for it 😄)

Reply:
I would love to use one Kuebconfig, but right now the code that deals with build clusters does assume that exactly. I think that adding support for multiple `--kubeconfig` flags in the future will be simple and we can tackle it at a later point.

Reply:
Which config do you use for each cluster? Given that you can add an arbitrary number of users, clusters and contexts of the two to a single kubeconfig file, why would we need multiple of these files?

Reply:
I think one context per cluster is the only sane way to do this and yes, one kubeconfig should be able to hold all of it.